src/claimtrie.cpp

```diff
@@ -1237,20 +1212,80 @@ bool CClaimTrieCache::recursiveComputeMerkleHash(CClaimTrieNode* tnCurrent, std:
     return true;
 }
 
-uint256 CClaimTrieCache::getMerkleHash() const
+bool CClaimTrieCache::recursiveComputeMerkleHashBinaryTree(CClaimTrieNode* tnCurrent, std::string sPos, bool forceCompute) const
+{
+    if (tnCurrent->empty()) {
+        cacheHashes[sPos] = _noChildrenOrClaims;
+        return true;
+    }
+
+    std::vector<uint256> childHashes;
+    for (nodeMapType::iterator it = tnCurrent->children.begin(); it != tnCurrent->children.end(); ++it) {
+        std::stringstream ss;
+        ss << it->first;
+        std::string sNextPos = sPos + ss.str();
+        if (forceCompute || dirtyHashes.count(sNextPos) != 0) {
```

Comment:
count look-up through whole container constant, better use find and compare against end.

Reply:
Again, both the stringstream and the count were in the previous method; I wasn't focused on those. If I do another round of changes on this story, I'll remove them though.

Reply:
Yeah, i saw that but it's good time to fix it.